libr/search/search.c

```diff
@@ -125,258 +131,172 @@ R_API int r_search_hit_new(RSearch *s, RSearchKeyword *kw, ut64 addr) {
 	return true;
 }
 
+// Supported search variants: overlap
 R_API int r_search_deltakey_update(void *_s, ut64 from, const ut8 *buf, int len) {
 	RListIter *iter;
-	unsigned char pch = 0;
 	int i, j, count = 0;
 	RSearch *s = (RSearch*)_s;
+	RSearchKeyword *kw;
 
-	for (i=0; i<len; i++) {
-		RSearchKeyword *kw;
-		r_list_foreach (s->kws, iter, kw) {
-			for (j=0; j<=kw->distance; j++) {
-				char ch = kw->bin_keyword[kw->idx[j]]; // signed char
-				ut8 ch2 = buf[i];
-				/* no icase in delta keys */
-				/* no binmask in delta keys */
-				/* no inverse support for delta keys */
-				if (pch+ch == ch2) {
-					kw->idx[j]++;
-					if (kw->idx[j] == kw->keyword_length) {
-						if (!r_search_hit_new (s, kw, (ut64)
-							from + i - kw->keyword_length + 1))
-							return -1;
-						kw->idx[j] = 0;
-						//kw->idx[0] = 0;
-						kw->distance = 0;
-						kw->count++;
-						count++;
-						//s->nhits++;
+	r_list_foreach (s->kws, iter, kw) {
+		ut8 *a = kw->bin_keyword;
+		for (i = 0; i + kw->keyword_length < len; i++) {
+			if ((ut8)(buf[i+1] - buf[i]) == a[0]) {
```

Comment:
https://github.com/radare/radare2-regressions/pull/959

I do not see how to make `/d` output one more byte for each hit.